### ISSUE TITLE
Parser: No error on function without body in TS

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#16683]
This is legal in a `.d.ts` file, but illegal in plain `.ts`:

```ts
function f(): void;
```

Oxc parser does not complain (and nor does semantic): [Oxc playground](https://playground.oxc.rs/#eNptU8Fu2zAM/ZVCpw3IgK7AgCI7De0CDOjWbh3Wyy60RLtKZckl6SRFkH8v5VpKDj3ZehTJx8envbFmadoxWvEpnrUfPi7PNsm7r/+jWZhklntDY8yf4KOYZQuBcWHaRD2IIFVECCJnuCKeUwBBd402AEGuzzW4ffSCPIDFCvUQu3A82tQPhHxMYZsGDQuN+fDSNymUk227+dphYQYgzsT2BneCkbWvjiis80AIafsHZaR4Owp7h6t58lIpt0Ta4B2QphZ0yvt3+SMKeS1oa4BRWYu334kSzahSyFpNFPT/RKq9GRn/QnOcSaB58E4ezfJiYTC62/bGRx3ShFbpDtpN5vjluXbzWaHfY5KjTGve3b8DP+fTHaVBexngTxHRodOauiev9Lqr1PeQYyEoqnP4OqzOsi3za3wLL1m7hsA+odzrzjS9XC4o9PjGfO6fmjVaeSAYtEYRNTeauH4TFbIZlSHSSZqqVW30ppcAdai2M8gX55+/aAFV8BpbzVklugrAvPIYXF0I7gYk32MUCGq8pL6ri9Fo76WiP1HAgagI89beN+zesJIddPdIEdRyhapNDjuc3kbMO65uXLNL+qpmNWNMMtUqSMCu3J2qFJtns5Q3oJOn4QY3WIs+IQ6/VOXinSlV/ZjCSr2ZMzZITeITLX3MO8ih+ndQ2E3yZbj+HTKeZQYWc3gFQvRz/w==)

It should: [TS playground](https://www.typescriptlang.org/play/?#code/GYVwdgxgLglg9mABMAFASgFyIG5xgEwG4AoIA)

Seems to only affect plain functions. Semantic does error on this in a `.ts` file:

```ts
class C {
  f(): string;
}
```

[Oxc playground](https://playground.oxc.rs/#eNptU8Fu2zAM/ZVApw3IgK7AgCI7De0CDOjWbh3WSy60TLtKZckj6SSFkX8v5VpKDj3ZehTJx8en0VizMtYD8+J6MW7CYtF8+LhasJAL7ddNOG6CWZpoVqOhIaSPd0HMqgHPuDRNpA5EkAoiBIETXBDH0YNgfYPah0BcDFyC+ycnyD1YLFAHofWno41dT8inFLax17DQkA4vXRV9Ptmmna8dl6YH4kRsNHgQDKx9dVZhnQe8j/s/KAOFu0HY1bgegk3McqXUEmmH90CamtEp79/Vj5DkYWdLgFFZi7PfiSLNqFJIWk0U9P9MqtEMjH+hOs0kUD26Wp7M6nJpMNR3za0LOqTxjdLttZvM8asL7aa78fh7iHKSacuHh3fg/+l0T7HXXgb4U0Cssdaauien9Nrr2HWQYt4rqnO4MqzOss/za3wPL0m7isA+ozzozjQ9X84odPjGfO4fqy1aeSTotUYWNTWauH4TFbIalCHSWZqqVWz0ppcAtai2M8iXF5+/aAFV8AYbzVlHuk7+XTv0dVkIHnok12EQ8Gq8qL4ri9Fo56SgP1GgBlER5q29b9jRpEfR6+6RAqjlMlUba2xxehsh7bi4cct11Oc1qxlClKlWRjy2+e5UJds8mSW/AZ089re4w1L0GbH/pSpn70yp6sfo1+rNlLFDqiKfaelC2kEKlb+jwvUkX4LL3zHhSWZgMcdXkN52ww==)

**Comment by overlookmotel** [comment]
Note: This is still legal in plain TS files:

```ts
function f(): boolean;
function f(): number;
function f(): string {
  return "yeah";
}
```

Whether to raise an error on the 1st statement cannot be known until you've parsed the next 2. For this reason, I imagine it may not be trivial to implement the check in parser.

**Comment by Boshen** [comment]
Search for `Diagnostics.Function_implementation_is_missing_or_not_immediately_following_the_declaration` in https://github.com/microsoft/TypeScript/blob/v5.9.3/src/compiler/checker.ts. Really complicated.

**Comment by Boshen** [comment]
Close as too complicated.
